pygmt/tests/test_config.py

```diff
@@ -48,27 +46,40 @@ def test_config_font_one():
     Specifically, this test only checks that `FONT_ANNOT_PRIMARY`,
     `FONT_ANNOT_SECONDARY`, `FONT_LABEL`, and `FONT_TITLE` are modified.
     """
-    fig = Figure()
+    fig_ref, fig_test = Figure(), Figure()
+
     with config(FONT="8p,red"):
-        fig.basemap(region=[0, 9, 0, 9], projection="C3/3/9c", T="mjTL+w4c+d4.5+l")
-    fig.basemap(T="mjBR+w5c+d-4.5+l")
-    return fig
+        fig_ref.basemap(R="0/9/0/9", J="C3/3/9c", Tm="jTL+w4c+d4.5+l")
+    fig_ref.basemap(Tm="jBR+w5c+d-4.5+l")
 
+    with config(FONT="8p,red"):
+        fig_test.basemap(
+            region=[0, 9, 0, 9], projection="C3/3/9c", compass="jTL+w4c+d4.5+l"
+        )
+    fig_test.basemap(compass="jBR+w5c+d-4.5+l")
```

Comment:
There is a weird thing going on here. The tests fail when using `fig_ref, fig_test = Figure(), Figure()`, but pass when structured like so:

```python
    fig_ref = Figure()
    with config(FONT="8p,red"):
        fig_ref.basemap(R="0/9/0/9", J="C3/3/9c", Tm="jTL+w4c+d4.5+l")
    fig_ref.basemap(Tm="jBR+w5c+d-4.5+l")

    fig_test = Figure()
    with config(FONT="8p,red"):
        fig_test.basemap(
            region=[0, 9, 0, 9], projection="C3/3/9c", compass="jTL+w4c+d4.5+l"
        )
    fig_test.basemap(compass="jBR+w5c+d-4.5+l")
```

Not sure if this has any relation to #733?

Reply:
Yes, the same reason.